test/specs/lib/felaSanitizeCssPlugin-test.ts

```diff
@@ -0,0 +1,84 @@
+import sanitizeCss from 'src/lib/felaSanitizeCssPlugin'
+
+const validStringStyle = {
+  display: 'block',
+  margin: '0 0 0 0',
+}
+
+const validNumericStyle = {
+  top: 0,
+}
+
+const invalidCssValue = 'rgba('
+
+const invalidCssStyle = {
+  color: invalidCssValue,
+}
+
+const assertCssPropertyValue = (value: string, isValid: boolean) => {
+  test(`assert that '${value}' is ${isValid ? 'valid' : 'invalid'}`, () => {
+    const sanitize = sanitizeCss()
+
+    const style = { display: value }
+    const sanitizedStyle = sanitize(style)
+
+    expect(sanitizedStyle).toEqual(isValid ? style : {})
+  })
+}
+
+const sanitize = sanitizeCss()
+
+describe('tests', () => {
+  test('should ensure there are no non-closed brackets in CSS property value', () => {
+    const style = {
+      ...validStringStyle,
+      ...invalidCssStyle,
+    }
+
+    expect(sanitize(style)).toEqual(validStringStyle)
+  })
+
+  test('should skip numeric CSS property values', () => {
+    expect(sanitize(validNumericStyle)).toEqual(validNumericStyle)
+  })
+
+  test('should recursively process nested objects', () => {
+    const style = {
+      ...validStringStyle,
+      '::before': {
+        ...validStringStyle,
+        ...invalidCssStyle,
+      },
+    }
+
+    expect(sanitize(style)).toEqual({
+      ...validStringStyle,
+      '::before': {
+        ...validStringStyle,
+      },
+    })
+  })
+
+  test('should skip excluded CSS props', () => {
+    const cssPropertyToExclude = 'myProperty'
+
+    const sanitize = sanitizeCss({
+      skip: [cssPropertyToExclude],
+    })
+
+    const style = {
+      validStringStyle,
+      [cssPropertyToExclude]: invalidCssValue,
+    }
+
+    expect(sanitize(style)).toEqual(style)
+  })
```

Comment:
Would encourage less abstraction in testing.  Consider this comparison:

```jsx
const validStringStyle = {
  display: 'block',
  margin: '0 0 0 0',
}

const invalidCssValue = 'rgba('

test('should skip excluded CSS props', () => {
  const cssPropertyToExclude = 'myProperty'

  const sanitize = sanitizeCss({
    skip: [cssPropertyToExclude],
  })

  const style = {
    validStringStyle,
    [cssPropertyToExclude]: invalidCssValue,
  }

  expect(sanitize(style)).toEqual(style)
})
```

Same without abstraction:

```jsx
test('should skip excluded CSS props', () => {
  const sanitize = sanitizeCss({ skip: ['myProperty'] })

  const style = {
    validStringStyle: { display: 'block', margin: '0 0 0 0' },
    myProperty: 'rgba(',
  }

  expect(sanitize(style)).toEqual(style)
})
```

One or two non-dry properties are OK if the trade off is readability.  Suggest always prioritizing "code is for humans" above the most optimal solution.

Reply:
yes, totally valid point, will rework